### PR TITLE
added feature with testing and readme updated 

### PR DESCRIPTION
**Pull request by vnajack**
for convert to lowercase, remove punctuation and whitespaces, and optional contraction replacement

### CODE REVIEW

**Comment by LoganParker** [comment]
closes #4

File: test.py [hunk 1]
```diff
@@ -1,9 +1,24 @@
...
+import preprocess as p
 
 class MainTestCase(unittest.TestCase):    
     def io_test(self):
-        self.assertEqual('Our team had a debate for best names for looping variables ? i won', read_text('text_files/test.txt'))
+        self.assertEqual('Our team had a debate for best names for looping variables ? i won', p.read_text('text_files/test.txt'))
+
+    def test_text_proc_replace_contractions(self):
+        test_string = "Hi   \nHow're\nYou?\nI'm well!\n Don't? Would've, I'll, she's, she'll!\n"
+        expected_string_no_conts = "hi how are you i am well do not would have i will she is she will"
```
Comment (LoganParker) [inline test.py:10]: These are hard to read, but I can't think of any other way of doing this, and it makes sense!

File: preprocess.py [hunk 2]
```diff
@@ -9,8 +10,53 @@ def read_text(filename: str) -> str:
...
+    """
+    Return a copy of the string with English contractions replaced.
+    WARNING: will change word count.
+    """
+    string = re.sub(r'\'s', ' is', string) 
+    string = re.sub(r'n\'t', ' not', string)
+    string = re.sub(r'\'ve', ' have', string)
+    string = re.sub(r'\'re', ' are', string)
+    string = re.sub(r'\'m', ' am', string)
+    string = re.sub(r'\'ll', ' will', string)
```
Comment (LoganParker) [inline preprocess.py:30]: this could probably be combined into one line using regex, but I think for clarity, it's ok to not do so!

**Review by LoganParker — APPROVED** [review]
Overall looks great! nothing that I think requires revision :)

File: test.py (hunk 1 above)
Comment (vnajack) [reply test.py:10]: I tried to pick a complicated test case full of new lines, capital letters, and contractions - I didn't try to make it make sense, but I agree that if I had a good example that actually made sense that would've been better!

**Review by edouarde1 — APPROVED** [review]
I like the examples you put into the test cases. They cover all of the pre-processing functions you wrote out.

File: preprocess.py (hunk 2 above)
Comment (mkuelker) [reply preprocess.py:30]: Yeah, this code is very readable.

**Review by mkuelker — APPROVED** [review]
This Looks Great! 👯 🔥

**Pull request merged**